Add documentation on how to debug the SageMaker orchestrator (#1810)
zenml-io/zenml: docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md

![Sagemaker Studio Pipelines](../../../.gitbook/assets/sagemakerUI.png)

### Debugging SageMaker Pipelines

If your SageMaker pipeline failed before the first ZenML step was started, the
ZenML run will not show up in the ZenML dashboard. In these cases, you need to 
[use the SageMaker UI](#sagemaker-ui) to view the respective error message and

> logs. To do so, open the corresponding pipeline in the SageMaker UI, then open

Reviewer: A lot of assumption that the person will know how to open the pipeline in the Sagemaker UI. Maybe just a bit more information / hints about where they can go to find this?

Author: That's shown in a lot of detail incl. screenshots in the previous section (#sagemaker-ui)

> If your SageMaker pipeline failed before the first ZenML step was started, the
> ZenML run will not show up in the ZenML dashboard. In these cases, you need to 
> [use the SageMaker UI](#sagemaker-ui) to view the respective error message and
> logs. To do so, open the corresponding pipeline in the SageMaker UI, then open
> its only execution, then click on the failed step in the pipeline graph and go
> to the 'Output' tab:

Reviewer 2: I personally find bullet points or numbered lists a better approach to describe series of steps but this paragraph is also great.
```suggestion
If your SageMaker pipeline encounters an error before the first ZenML step starts, the ZenML run will not appear in the ZenML dashboard. In such cases, use the [SageMaker UI](#sagemaker-ui) to review the error message and logs. Here's how:

Open the corresponding pipeline in the SageMaker UI.
Open its execution.
Click on the failed step in the pipeline graph.
Go to the 'Output' tab to access logs.
```

Author: Great suggestions 👍

![SageMaker Studio Logs](../../../.gitbook/assets/sagemaker-logs.png)

Alternatively, you can use 
[Amazon CloudWatch](https://aws.amazon.com/cloudwatch/) to get an even more 
detailed view of all the log messages that were written during the execution of 
your SageMaker pipeline. To do so, search for 'CloudWatch' in the search bar of

> your AWS terminal, navigate to 'Logs > Log groups' and open the

Author: ```suggestion
your AWS console, navigate to 'Logs > Log groups' and open the
```

> Alternatively, you can use 
> [Amazon CloudWatch](https://aws.amazon.com/cloudwatch/) to get an even more 
> detailed view of all the log messages that were written during the execution of 
> your SageMaker pipeline. To do so, search for 'CloudWatch' in the search bar of
> your AWS console, navigate to 'Logs > Log groups' and open the
> '/aws/sagemaker/ProcessingJobs' log group. Here you will be able to see one log
> stream per step for each of your SageMaker pipeline executions:

Reviewer 2: ```suggestion
Alternatively, for a more detailed view of log messages during SageMaker pipeline execution, consider using [Amazon CloudWatch](https://aws.amazon.com/cloudwatch/):

Search for 'CloudWatch' in the AWS console search bar.
Navigate to 'Logs > Log groups.'
Open the '/aws/sagemaker/ProcessingJobs' log group.
Here, you can find log streams for each step in your SageMaker pipeline executions.
```
